cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/etl/realtime/ETLWorker.java

```diff
@@ -267,33 +273,38 @@ public void run(DatasetContext context) throws Exception {
         continue;
       }
 
-      // For each object emitted by the source, invoke the transformExecutor and collect all the data
-      // to be persisted in the sink.
-      for (Object sourceData : sourceEmitter) {
-        try {
-          TransformResponse transformResponse = transformExecutor.runOneIteration(sourceData);
-          while (transformResponse.getEmittedRecords().hasNext()) {
-            dataToSink.add(transformResponse.getEmittedRecords().next());
-          }
+      getContext().execute(new TxRunnable() {
```

Comment:
We may want some way to not have to wrap every transform operation in a transaction, but not sure.